spec/controllers/two_factor_authentication/otp_expired_controller_spec.rb

```diff
@@ -0,0 +1,22 @@
+require 'rails_helper'
+
+describe TwoFactorAuthentication::OtpExpiredController do
+  describe '#show' do
+    it 'renders the page on sign in' do
+      sign_in_before_2fa
+
+      get :show
+
+      expect(response).to render_template(:show)
+    end
+
+    it 'renders the page when adding a new one time code method' do
+      user = build(:user, otp_delivery_preference: 'voice')
+
+      stub_sign_in_before_2fa(user)
+      get :show
+
+      expect(response).to render_template(:show)
+    end
```

Comment:
What would be the difference we expect between "on sign in" and "when adding a new one time code method" ?

It may also be good to assert that the expected view assigns are assigned as expected (i.e. `@otp_delivery_preference`).

Reply:
My thought behind this was that I was considering the paths of when a user has an existing phone number -> enter their code -> code vs when a user is creating a new phone number -> enters new phone number -> enter code -> expired.

As for testing `@otp_delivery_preference` it will say that `otp_delivery_preference` is undefined and could not access to that variable, hence while I hard-coded it in the user.

Reply:
By testing, I was thinking something which asserts the expected value, like...

```rb
expect(assigns(:otp_delivery_preference)).to eq('something')
```